negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationConsumerService.java

```diff
@@ -105,7 +107,9 @@ public void handleAgreement(ContractAgreementMessage contractAgreementMessage) {
     			.callbackAddress(contractAgreementMessage.getCallbackAddress())
     			.assigner(contractNegotiation.getAssigner())
     			.state(ContractNegotiationState.AGREED)
+    			.role(contractNegotiation.getRole())
```

Comment:
Can this one be changed with withNewContractNegotiationState(ContractNegotiationState.AGREED)?

Reply:
Similar like previous
agreement(contractAgreementMessage.getAgreement())
not from contractNegotiation